packages/plugin-e2e/src/models/pages/DashboardPage.ts

```diff
@@ -110,4 +110,22 @@ export class DashboardPage extends GrafanaPage {
   async refreshDashboard() {
     await this.ctx.page.getByTestId(this.ctx.selectors.components.RefreshPicker.runButtonV2).click();
   }
+
+  /**
+   * Saves the dashboard
+   */
+  async saveDashboard(title?: string): Promise<string | undefined> {
```

Comment:
Having e2e tests that rely on server state can be troublesome. So far, we've been able to avoid writing tests that changes server state during test execution and there's no plugin-e2e APIs that changes server state. 

Can you elaborate a bit on why this is necessary? If this is related to testing variables, isn't it enough to add the variable, close settings and verify it's working as expected?

Reply:
I can rewrite my test to do so, but it becomes more challenging to add multiple template variables by just closing the settings as the `VariableEditPage` model will be redirected to the variable edit page for a new dashboard each time it is called. This makes it difficult to test template variables that depend on eachother.

Reply:
In these situations where you have cascading variable dependencies, I recommend that you provision a dashboard that have all these variables configured already. Then you can use `gotoVariableEditPage` and add a new variable that refers to an already existing one. 

You can checkout our guide on [test isolation](https://grafana.com/developers/plugin-tools/e2e-test-a-plugin/setup-resources#test-isolation). The Playwright docs also has a good [article](https://playwright.dev/docs/browser-contexts#what-is-test-isolation) on this subject.

Reply:
That's a good shout! I will actually split these tests into ones that just create the template variables individually and a separate one for utilising the variables. Would you prefer if I remove the functionality for dashboard saving?

Reply:
Sounds good! Yeah I think we can remove the saving functionality, thank you!